### DISCUSSION TITLE
Media Types with parameters are excluded when choosing a default renderer

### DISCUSSION THREAD

**Discussion by Roanmh** [encode/django-rest-framework#9564] [Potential Issue]
The documentation states

> If a client underspecifies the representations it can accept, such as sending an Accept: */* header, or not including an Accept header at all, then REST framework will select the first renderer in the list to use for the response

but when the renderer specifies parameters (other than `q`), such as the [JSON-API `ext` parameter](https://jsonapi.org/format/#media-type-parameter-rules). Then an [exact match](https://github.com/encode/django-rest-framework/blob/master/rest_framework/utils/mediatypes.py#L53) of parameters (unordered) is required. Otherwise a 406 "Could not satisfy the request Accept header." is returned.

Is this behavior desired or is this a bug?

**Comment by lovelydinosaur** [comment]
Seems reasonable to me...

* `Accept: application/vnd.api+json` will require a JSON API renderer.
* `Accept: application/vnd.api+json; ext="https://jsonapi.org/ext/version"` will require a JSON API renderer matching that extension.

**Reply by Roanmh** [reply]
I don't think my question was clear:

For the case `Accept: application/vnd.api+json` is a renderer with `application/vnd.api+json; ext="https://jsonapi.org/ext/version"` valid? i.e. does the lack of parameters imply acceptance of all parameters or no parameters? If not, I don't think there is a way to specify acceptance of any parameter value right?

Or especially the wild card case, `Accept: */*`, is that renderer valid? If a client can specify any media type, should a client also be able to specify any parameter values?

**Reply by auvipy** [reply]
should we handle this in DRF core or something like https://github.com/django-json-api/django-rest-framework-json-api/blob/main/rest_framework_json_api/renderers.py ?
